be/src/exec/vectorized/aggregate/aggregate_blocking_node.cpp

```diff
@@ -168,10 +169,15 @@ std::vector<std::shared_ptr<pipeline::OperatorFactory> > AggregateBlockingNode::
     if (agg_node.need_finalize) {
         // If finalize aggregate with group by clause, then it can be paralized
         if (agg_node.__isset.grouping_exprs && !_tnode.agg_node.grouping_exprs.empty()) {
-            std::vector<ExprContext*> group_by_expr_ctxs;
-            Expr::create_expr_trees(_pool, _tnode.agg_node.grouping_exprs, &group_by_expr_ctxs);
-            operators_with_sink =
-                    context->maybe_interpolate_local_shuffle_exchange(operators_with_sink, group_by_expr_ctxs);
+            // The san operator followed by an aggregate operator indicates the aggregating key is the same as
+            // the bucket key of this table, so we needn't local shuffle for this case.
+            auto* source_op = operators_with_sink[0].get();
+            if (operators_with_sink.size() > 1 || typeid(*source_op) != typeid(pipeline::ScanOperatorFactory)) {
```

Comment:
typid cannot judge derive class or base class, you could replace it with dynamic_cast